perllib/FixMyStreet/App/Controller/Waste.pm

```diff
@@ -835,6 +836,57 @@ sub process_request_data : Private {
     return 1;
 }
 
+sub ggw_transfer_subscription : Private {
+    my ($self, $c, $form) = @_;
+    my $data = $form->saved_data;
+
+    # Get the current subscription for the old address
+    my $old_property_id = $data->{previous_ggw_address}[0]->{value};
+    #$c->forward('get_original_sub', ['', $old_property_id]);
+
+    my $base = {};
+    $base->{name} = $c->get_param('name');
+    $base->{email} = $c->get_param('email');
+    $base->{phone} = $c->get_param('phone');
+
+    # Cancel the old subscription
+    my $cancel = { %$base };
+    $cancel->{category} = 'Cancel Garden Subscription';
+    $cancel->{title} = 'Garden Subscription - Cancel';
+    $cancel->{address} = $data->{previous_ggw_address}[0]->{label};
+    my $now = DateTime->now->set_time_zone(FixMyStreet->local_time_zone);
+    my $end_date_field = $c->cobrand->call_hook(alternative_backend_field_names => 'Subscription_End_Date') || 'Subscription_End_Date';
+    $c->set_param($end_date_field, $now->ymd);
+    $c->set_param('property_id', $old_property_id);
+    $c->set_param('uprn', $data->{transfer_old_ggw_sub}{transfer_uprn});
+    $c->forward('setup_garden_sub_params', [ $cancel, undef ]);
+    $c->forward('add_report', [ $cancel ]) or return;
+    $c->stash->{report}->confirm;
+    $c->stash->{report}->update;
+
+    # Create a report for it for the new address
+    my $new = { %$base };
+    $new->{category} = 'Garden Subscription';
+    $new->{title} = 'Garden Subscription - New';
+    my $old_subscription_bin_data = $data->{transfer_old_ggw_sub}->{ServiceTasks}->{ServiceTask}->{Data}->{ExtensibleDatum}[0]->{ChildData}->{ExtensibleDatum};
+    for my $bins (@{$old_subscription_bin_data}) {
+        if ($bins->{DatatypeName} eq 'Quantity') {
+            $new->{current_bins} = $bins->{Value};
+        } elsif ($bins->{DatatypeName} eq 'Container Type') {
+            $new->{current_type} = $bins->{Value};
+        }
+    }
+    #my $expiry = $data->{transfer_old_ggw_sub}->{ServiceTasks}->{ServiceTask}->{ServiceTaskSchedules}->{ServiceTaskSchedule}[0]->{EndDate}->{DateTime};
```

Comment:
Yes, so store this before and look it up here, set it on the field (which is `End Date` also, yes) - will need adding to the test's category extra data fields list when set up, then it should be present in your test.